examples/graphene_apps/cppopenvino/process_wo.cpp

```diff
@@ -0,0 +1,132 @@
+        return 1;
+    }
+    // Create work order processor.
+    ProcessWorkOrder process_wo(in_image_dir, out_image_dir);
+    // Load SSD model.
+    int ret = process_wo.load_model(model_file);
+    if (ret != 0) {
+        std::cout << "Loading Openvino SSD model failed" << std::endl;
+        return 1;
+    }
+    // Listen for requests.
+    ZmqListener zmq_listener(zmq_url, process_wo);
+    std::cout<< "Start listening to ZMQ URL: ";
+    std::cout << zmq_url <<" for OV work orders" << std::endl;
+    zmq_listener.start_zmq_listener();
+
+    return 0;
+}
+
+/**
+*  ProcessWorkOrder constructor.
+*
+*  @param in_image_dir Path where input images are kept.
+*  @param out_image_dir Path where output images are kept.
+*/
+ProcessWorkOrder::ProcessWorkOrder(const std::string in_image_dir, 
+                                   const std::string out_image_dir)
+        :in_image_dir(in_image_dir),out_image_dir(out_image_dir) {  
+}
+
+/**
+*  Load model which will be used for object detection.
+*
+*  @param modelFile Model file name.
+*  @return 0 in case of success. Other values indicate failure.
+*/
+int ProcessWorkOrder::load_model(const std::string &modelFile) {
+    int ret = ssd.load_model(modelFile);
+    return ret;
+}    
+
+/**
+*  Handles openvino inference work order.
+*
+*  @param in_image_dir Path where input imaages are kept.
+*  @param out_image_dir Path where output images are kept.
+*  @return output message as string.
+*/
+std::string ProcessWorkOrder::process_work_order(std::string &msg) {
+    std::string output;
+    std::cout << "Message: " << msg << std::endl;
+    json j = json::parse(msg);
+    if ( j.find("workloadId") == j.end() ) {
+        output = "Error: Json doesn't contain workloadId";
+        std::cout << output << std::endl;
+        return output;
+    }
+    if ( j.find("params") == j.end() ) {
+        output = "Error: Json doesn't contain input params";
+        std::cout << output << std::endl;
+        return output;
+    }
+    std::string workload = j["workloadId"];
+    std::string in_image_name = j["params"];
+    std::cout << "workload :" << workload << std::endl;
+    // Process ov-inference workload.
+    if(workload.compare("ov-inference") == 0) {
+        std::string in_image_path = in_image_dir + in_image_name;
+        // Create output bmp image name.
+        // TODO: Do more error checks.
+        size_t dot_index = in_image_name.find_last_of(".");
+        std::string out_image_name = in_image_name.substr(0, dot_index) + ".bmp";
+        std::string out_image_path = out_image_dir + out_image_name;
+        int ret = ssd.object_detection(in_image_path, out_image_path);
+        if(ret == 0) {
+            output = "Openvino Success: Generated output file: " + out_image_name;
+        } else {
+            output = "Openvino Error: Could't execute OpenVino workorder";
+        }
+    } else {
+        output = "Error: Unsupported workload";
```

Comment:
We could preferably returns objects with return codes and message. But not really an issue at this stage.

Reply:
For now the function is returning string. The json conversion happens in Avalon python worker and this application is independent of Avalon worker.